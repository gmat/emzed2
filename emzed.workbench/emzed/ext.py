import pkg_resources

for ep in pkg_resources.iter_entry_points("emzed_plugin", name="package"):
    print ep
    package = ep.load()
    exec "%s=package" % package.__name__
    del package
    del ep
del pkg_resources

