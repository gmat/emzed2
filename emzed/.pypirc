[distutils]
index-servers = 
  local

[local]
repository: http://127.0.0.1:8080
username:uschmitt
password:pillepalle

