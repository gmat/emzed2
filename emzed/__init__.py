# encoding:latin-1

from version import version as __version__


