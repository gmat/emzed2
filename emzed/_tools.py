
def is_started_from_cmdline():

    import inspect
    info = inspect.stack()[-1][1]
    result = False
    if info != "<string>":
        if info.endswith("emzed.workbench"):
            result = True
    return result


def gui_running():
    from PyQt4.QtGui import QApplication
    return QApplication.instance() is not None


def is_first_start():
    from emzed.core.config import _UserConfig
    import os
    return not os.path.exists(_UserConfig.config_file_path())
