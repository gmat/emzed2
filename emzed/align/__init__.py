from rt_align import rtAlign
from mz_align import mzAlign
