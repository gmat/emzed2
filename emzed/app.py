
import pkg_resources
reload(pkg_resources)  # migh be loaded already and is not up to date

for ep in pkg_resources.iter_entry_points("emzed_package", name="main"):
    runner = ep.load()
    print "$$$ loaded", runner
    exec "%s=runner" % ep.module_name.split(".")[0]
    del ep
    del runner

del pkg_resources

