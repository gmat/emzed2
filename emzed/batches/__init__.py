from peak_picker import *

# remove clutter
try:
    del peak_picker
except: 
    pass
