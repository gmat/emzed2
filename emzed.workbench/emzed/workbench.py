def run():
    print 42

def main():
    import os
    print os.system("pip install -i http://127.0.0.1:3142/root/dev/+simple emzed.core")
    import emzed.core
    print emzed.core



