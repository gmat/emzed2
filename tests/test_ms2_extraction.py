# encoding: utf-8
from __future__ import print_function

import emzed
import pytest


@pytest.fixture
def peaks(path):
    peaks = emzed.io.loadTable(path("data", "peaks_for_ms2_extraction.table"))

    # in order to reduce output for regtest:
    peaks.dropColumns("feature_id", "intensity", "quality", "fwhm", "z", "source")

    # additionally reduces output:
    return peaks.filter((peaks.id <= 6) | (peaks.id > 17))


@pytest.fixture
def peakmap(path):
    return emzed.io.loadPeakMap(path("data", "peaks_for_ms2_extraction.mzXML"))


def check(peaks, peakmap, regtest, mode):
    print(file=regtest)
    print("MODE=", mode, file=regtest)
    print(file=regtest)
    emzed.utils.attach_ms2_spectra(peaks, peakmap, mode=mode)

    def mz_range(spectra):
        mzs = [mz for s in spectra for mz in s.peaks[:, 0]]
        return max(mzs) - min(mzs)

    def energy(spectra):
        iis = [ii for s in spectra for ii in s.peaks[:, 1]]
        return sum(i * i for i in iis)

    peaks.addColumn("ms2_mz_range", peaks.spectra_ms2.apply(mz_range), type_=float)
    peaks.addColumn("ms2_energy", peaks.spectra_ms2.apply(energy), type_=float, format_="%.2e")
    peaks.addColumn("ms2_spec_count", peaks.spectra_ms2.apply(len), type_=int, format_="%d")
    peaks.setColFormat("spectra_ms2", None)

    print(peaks, file=regtest)


def test_mode_is_intersection(peaks, peakmap, regtest):
    check(peaks, peakmap, regtest, "intersection")


def test_mode_is_union(peaks, peakmap, regtest):
    check(peaks, peakmap, regtest, "union")


def test_mode_is_max_range(peaks, peakmap, regtest):
    check(peaks, peakmap, regtest, "max_range")


def test_mode_is_max_energy(peaks, peakmap, regtest):
    check(peaks, peakmap, regtest, "max_energy")


def test_mode_is_all(peaks, peakmap, regtest):
    check(peaks, peakmap, regtest, "all")
