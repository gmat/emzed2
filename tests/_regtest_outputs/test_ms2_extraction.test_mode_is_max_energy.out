
MODE= max_energy

id       mz        mzmin      mzmax      rt       rtmin    rtmax    peakmap  num_spectra_ms2 ms2_extraction_info ms2_mz_range ms2_energy
int      float     float      float      float    float    float    None     int             str                 float        float
------   ------    ------     ------     ------   ------   ------   ------   ------          ------              ------       ------
0        315.17123  315.17105  315.17136 21.32m   21.17m   24.49m   -        4               max_energy          258.11512    2.05e+15
1        316.17476  316.17447  316.17627 21.32m   21.06m   22.43m   -        0               max_energy          -            -
2        317.17663  317.17508  317.17810 21.32m   20.76m   21.70m   -        0               max_energy          -            -
3        318.17894  318.17874  318.17917 21.32m   21.23m   21.49m   -        0               max_energy          -            -
4        319.18149  319.16556  319.19073 21.32m   21.26m   21.93m   -        0               max_energy          -            -
5        263.20013  263.19992  263.20050 18.98m   18.86m   21.46m   -        4               max_energy          215.97846    3.13e+14
6        264.20343  264.20328  264.20374 18.98m   18.89m   19.68m   -        0               max_energy          -            -
18       237.15480  237.15474  237.15488 20.00m   19.97m   20.12m   -        0               max_energy          -            -
19       238.14875  238.14859  238.14981 20.00m   19.94m   20.38m   -        1               max_energy          185.11055    2.20e+11

